### PR TITLE
Fix Issue 16062 - Add 'clear' method to OutBuffer (std.outbuffer)

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

File: std/outbuffer.d [hunk 1]
```diff
@@ -202,6 +202,14 @@ class OutBuffer
             fill0(alignsize - nbytes);
     }
 
+    /*********************************
+     * Clear the data in the outbuffer.
+     */
```
Comment (wilzbach) [inline std/outbuffer.d]: FYI: `outbuffer` is a very old module, the more modern equivalent docstring would `/// Clear ...` ;-)

File: std/outbuffer.d [hunk 2]
```diff
@@ -380,6 +388,10 @@ class OutBuffer
     buf.write("world"[]);
     buf.printf(" %d", 62665);
     assert(cmp(buf.toString(), "hello world 62665") == 0);
+
+    buf.clear();
```
Comment (wilzbach) [inline std/outbuffer.d]: Test `buf.toString` here as well

**Comment by RazvanN7** [comment]
@wilzbach done

File: std/outbuffer.d [hunk 3]
```diff
@@ -368,18 +374,24 @@ class OutBuffer
...
 
+///
 @safe unittest
 {
     import std.string : cmp;
 
     OutBuffer buf = new OutBuffer();
 
     assert(buf.offset == 0);
-    buf.write("hello"[]);
```
Comment (wilzbach) [inline std/outbuffer.d]: I assume this is a relict from old times? In this case, it's even sliced internally:

```d
void write(const(ubyte)[] bytes)
{
    reserve(bytes.length);
    data[offset .. offset + bytes.length] = bytes[];
    offset += bytes.length;
}
...
void write(in char[] s) @trusted             /// ditto
{
    write(cast(ubyte[]) s);
}
```

Comment (RazvanN7) [inline std/outbuffer.d]: yep

**Pull request merged**